src/client/executor.rs

```diff
@@ -543,6 +543,9 @@ impl Client {
                     TransactionState::Starting => {
                         cmd.set_start_transaction();
                         cmd.set_autocommit();
+                        if session.causal_consistency() {
```

Comment:
The spec requires that the first operation in a causally consistent transaction have `afterClusterTime` attached, even if it's a write; previously the driver would only do that if it was a read.  This wasn't caught in the legacy tests because we had a skip in there pre-dating the implementation of causal consistency that never got removed :)